File: LeetCode/Python Solutions/Pseudo-Palindromic Paths in a Binary Tree/palindromic.py
class Solution:
    def pseudoPalindromicPaths(self, root: TreeNode) -> int:
        def dfs(node, path):
            if not node:
                return 0

            # Toggle the bit for the current node's value
            path ^= 1 << node.val
            
            # If it's a leaf node, check if the path is pseudo-palindromic
            count = 0
            if not node.left and not node.right:
                # Check if at most one bit is set in 'path'
                count = 1 if (path & (path - 1)) == 0 else 0
            
            # Continue DFS on left and right children
            return count + dfs(node.left, path) + dfs(node.right, path)

        return dfs(root, 0)

File: LeetCode/Java Solutions/Pseudo-Palindromic Paths in a Binary Tree/palindromic.java
class Solution {
    public int pseudoPalindromicPaths(TreeNode root) {
        return dfs(root, 0);
    }

    private int dfs(TreeNode node, int path) {
        if (node == null) {
            return 0;
        }

        // Toggle the bit for the current node's value
        path ^= 1 << node.val;

        int count = 0;
        // If it's a leaf node, check if the path is pseudo-palindromic
        if (node.left == null && node.right == null) {
            count = (path & (path - 1)) == 0 ? 1 : 0;
        }

        // Continue DFS on left and right children
        return count + dfs(node.left, path) + dfs(node.right, path);
    }
}
